modules/coredns/collect.go

```diff
@@ -61,14 +83,50 @@ func (cd *CoreDNS) collect() (map[string]int64, error) {
 	return stm.ToMap(mx), nil
 }
 
-// Summary
+func (cd *CoreDNS) updateVersionDependentMetrics(raw prometheus.Metrics) {
+	if cd.version == nil {
+		if version := cd.parseVersion(raw); version != nil {
+			cd.version = version
+			if cd.version.LTE(version169) {
+				cd.metricNames.panicCountTotal = metricPanicCountTotal169orOlder
+				cd.metricNames.requestCountTotal = metricRequestCountTotal169orOlder
+				cd.metricNames.requestTypeCountTotal = metricRequestTypeCountTotal169orOlder
+				cd.metricNames.responseRcodeCountTotal = metricResponseRcodeCountTotal169orOlder
+			} else {
+				cd.metricNames.panicCountTotal = metricPanicCountTotal170orNewer
+				cd.metricNames.requestCountTotal = metricRequestCountTotal170orNewer
+				cd.metricNames.requestTypeCountTotal = metricRequestTypeCountTotal170orNewer
+				cd.metricNames.responseRcodeCountTotal = metricResponseRcodeCountTotal170orNewer
+			}
+		}
+	}
+	cd.skipVersionCheck = true
+}
```

Comment:
neat: let's use early return, that is always a good idea in my opinion. And I think we don't need to check "if cd.version == nil", that func states that it updates the version, so let's do it (it will be only once anyway because there is "if !cd.skipVersionCheck" check in the collect).